bmp280/bmp280.go

```diff
@@ -0,0 +1,240 @@
+package bmp280
+
+import (
+	"machine"
+	"time"
+)
+
+// OversamplingMode is the oversampling ratio of the temperature or pressure measurement.
+type Oversampling uint
+
+// Mode is the Power Mode.
+type Mode uint
+
+// Standby is the inactive period between the reads when the sensor is in normal power mode.
+type Standby uint
+
+// Filter unwanted changes in measurement caused by external (environmental) or internal changes (IC).
+type Filter uint
+
+// Device wraps an I2C connection to a BMP280 device.
+type Device struct {
+	bus         machine.I2C
+	Address     uint16
+	cali        calibrationCoefficients
+	Temperature Oversampling
+	Pressure    Oversampling
+	Mode        Mode
+	Standby     Standby
+	Filter      Filter
+}
+
+type calibrationCoefficients struct {
+	// Temperature compensation
+	t1 uint16
+	t2 int16
+	t3 int16
+
+	// Pressure compensation
```

Comment:
what do you think about using an array instead? 
```go
p [8]uint16 
```

Reply:
I would prefer to not change this, since the compensation formula from the datasheet is already hard to read and I don't want to add another layer of confusion by changing everything from say t1 to t[0] and p3 to p[2].